src/main/java/servlets/ReceiptAnalysis.java

```diff
@@ -288,6 +290,45 @@ private static boolean isDate(String token) {
     return dateRegex.matcher(token).matches();
   }
 
+  /**
+   * Checks the raw text in the builder for prices that can be parsed. The largest price found, if
+   * it exists, is added to the builder.
+   */
+  private static void checkForParsablePrices(AnalysisResults.Builder analysisBuilder) {
+    Stream<String> prices =
+        getTokensFromRawText(analysisBuilder.getRawText()).filter(ReceiptAnalysis::isPrice);
+    // Assume that the largest price on the receipt is the total price.
+    double largestPrice = prices.mapToDouble(ReceiptAnalysis::parsePrice)
+                              .reduce(Double.NEGATIVE_INFINITY, Double::max);
+
```

Comment:
The `Stream.max()` method can be used for this special case of reduction.

https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#max-java.util.Comparator-

Reply:
The issue with that option is that I'd have to check for both an empty `Optional` and `Double.NEGATIVE_INFINITY`, hence why I used reduce with an identity.